# Taxonomic binning of metagenome samples generated by next-generation sequencing technologies {#sec:full_bib}

**J. Dröge**^1,2^ and **A. C. McHardy**^1,2\*^

^1^Max Planck Research Group for Computational Genomics & Epidemiology, Max Planck Institute for Informatics, Campus E1 4, 66123 Saarbrücken, Germany

^2^Department of Algorithmic Bioinformatics, Heinrich Heine University Düsseldorf, Institute for Computer Science, Universitätsstraße 1, 40225 Düsseldorf, Germany

----

**This is a pre-copyedited, author-produced version of an article accepted for publication in *[Briefings in Bioinformatics](https://bib.oxfordjournals.org)* following peer review. This article version has been adapted to the thesis layout. The original article is available online by DOI [10.1093/bib/bbs031](https://doi.org/10.1093/bib/bbs031).**

## Abstract

Metagenome research uses random shotgun sequencing of microbial community DNA to study the genetic sequences of its members without cultivation. This development has been strongly supported by improvements in sequencing technologies, which have rendered sequencing cheaper than before. As a consequence, downstream computational analysis of metagenome sequence samples is now faced with large amounts of complex data. One of the essential steps in metagenome analysis is reconstruction of draft genomes for populations of a community or of draft 'pan-genomes' for higher-level clades. ‘Taxonomic binning’ corresponds to the process of assigning a taxonomic identifier to sequence fragments, based on information such as sequence similarity, sequence composition or read coverage. This is used for draft genome reconstruction, if sequencing coverage is insufficient for reconstruction based on assembly information alone. Subsequent functional and metabolic annotation of draft genomes allows a genome-level analysis of novel uncultured microbial species and even inference of their cultivation requirements.

## Introduction

The application of genome sequencing technologies to the study of an entire community of microbial organisms, as opposed to a clonal culture of an individual isolate strain, is known as metagenomics [@SimonMetagenomic2011; @KuninBioinformatician2008]. Such analysis allows one to determine genome sequence information for a vast portion of the microbial world for which cultivation conditions are unknown or difficult to reproduce under laboratory conditions [@AmannPhylogenetic1995; @HugenholtzExploring2002]. Even the first metagenome studies, investigating the Sargasso Sea [@VenterEnvironmental2004] and Minnesota farm soil [@TringeComparative2005], were able to demonstrate the enormous potential of the microbial world to serve as a treasure trove of genes with novel functionalities, as these studies resulted in the discovery of many thousands of new gene sequences that were only remotely similar to genes of known function. They also revealed the unexpected complexity of microbial communities in terms of the number of taxa contained therein. Since then, much research has explored microbial ecosystems, soil, aquatic and host-associated, in more detail [@WoykeSymbiosis2006; @SuenInsect2010; @TurnbaughOrganismal2010; @WarneckeMetagenomic2007; @MackelprangMetagenomic2011], and has revealed a great wealth of novel genetic information from microbial species that are only distantly related to well studied model organisms.

Both amplicon sequencing and random shotgun sequencing of microbial communities are sometimes referred to as metagenomics. Amplicon sequencing, or environmental tag sequencing, is used to determine the taxonomic composition and phylogenetic structure of a microbial community. In amplicon sequencing, informative marker regions of the genomes from a microbial community are amplified by polymerase chain reaction, and used as a proxy to determine which phylotypes or operational taxonomic units (OTUs) are present in a microbial community, and their relative abundance. Commonly used markers regions are the ribosomal genes [@HuseExploring2008] and the ITS (internal transcribed spacer) region [@JeewonDetection2007], which is positioned between ribosomal genes.  In terms of numbers and the evolutionary closeness of the distinct species present, microbial community profiles can be correlated across environments and communities, linked to environmental parameters. They can be indicative of the presence of genes that are relevant for particular metabolic functionalities [@FuhrmanMicrobial2009], given that the respective genes are already known. However, the gene inventory and the encoded functionality of most microbial species are largely unknown and may also vary considerably between strains.

Shotgun sequencing can be used to study the genetic information of microbial communities by sequencing DNA that has been extracted and randomly sheared into smaller fragments. Even though subject to different technology-dependent biases, this procedure allows  functional and process-level characterization of microbial communities as a whole and the reconstruction of draft genome sequences for individual community members.

## Next-generation sequencing technologies

DNA sequencing technologies have rapidly advanced over the last five years and these developments have substantially shaped the way metagenome research is performed. Post-Sanger sequencing technologies are commonly referred to as next-generation sequencing (NGS) [@MardisImpact2008; @MetzkerSequencing2009]. In comparison to Sanger sequencing, NGS methods can sequence DNA more quickly and at lower cost through massive parallelization. This is generally achieved by amplification and fixation of millions of individual template molecules or their enzyme counterparts on a solid phase prior to sequencing. While Sanger sequencing results in read lengths of around 800 bp, the commercially available NGS technologies ([@tbl:bib_sequencing_technologies]) currently generate reads of approximately 50–75 bp (Applied Biosciences/Life Technologies – SOLiD), 75–150 bp (Solexa/Illumina – Sequencing by Synthesis), 100–200 bp (IonTorrent/Life Technologies – Semiconductor Chip Sequencing) and 550–1000 bp (454/Roche – Pyrosequencing). The upcoming generation [@SchadtWindow2010; @ThompsonProperties2011] of sequencers using single molecule sequencing produces read lengths of over 1 kb (PacBio, SMRT, 15–20% assumed error rate [@SchadtWindow2010]) and of 5–10 kb (Oxford Nanopore technology, 5% assumed error rate). Besides different read lengths and amounts of sequence data produced, each technology has a characteristic profile of sequencing errors, resulting from the technology-specific preparation and detection procedures. The choice of an appropriate sequencing technology depends on the scientific questions asked. For instance, while an 80 bp read is sufficient to cover a hyper-variable region in the 16S gene [@HuseExploring2008] for analysis of microbial community composition, de novo recovery of draft microbial genome sequences by taxonomic binning from a complex organismal mixture requires substantially longer reads or higher sequencing depth and sequencing of short paired reads [@TurnbaughOrganismal2010; @MackelprangMetagenomic2011; @HessMetagenomic2011; @IversonUntangling2012].


+---------------------------------------------------+-------------+--------------------------+---------------------+----------------+-------------+
| Manufacturer                                      | Length (bp) | Through-                 | Normalized          | Through-       | Time per run|
| & technology                                      |             | put\*\*\*                | throughput\* (Mb/h) | put scale\*\*  |             |
+===================================================+=============+==========================+=====================+================+=============+
| Solexa/ Illumina Sequencing by Synthesis          |   100       | 300 Gb/8.5 d             | 1,500               | 10^4^          | 8.5 d       |
|                                                   |   – 150     | – 600 Gb/11 d            | – 2,300             |                | – 11 d      |
+---------------------------------------------------+-------------+--------------------------+---------------------+----------------+-------------+
| Life Technologies/ Applied Biosystems SOLiD       |   50        | 7 Gb/d                   | 300                 | 10^3^          | 2 d         |
|                                                   |   – 75      | – 20 Gb/d                | – 800               | - 10^4^        | – 7 d       |
+---------------------------------------------------+-------------+--------------------------+---------------------+----------------+-------------+
| Life Technologies/ Ion Torrent                    |   100       | 10 Mb/2 h                | 5                   | 10^1^          | 2 h         |
|                                                   |   – 200     | – 1 Gb/2 h               | – 500               | - 10^3^        |             |
+---------------------------------------------------+-------------+--------------------------+---------------------+----------------+-------------+
| Roche/ 454 Pyrosequencing                         |  550        | 450 Mb/10 h              | 30                  | 10^2^          | 10 h        |
|                                                   |  – 1000     | – 700 Mb/23 h            | – 45                |                | – 23 h      |
+---------------------------------------------------+-------------+--------------------------+---------------------+----------------+-------------+
| Life Technologies Capillary Sanger sequencing     |   600       | 690 Kb/d                 | 0.029               | 10^0^          | ~ 7 h [15]  |
|                                                   |   – 900     | – 2,100 Kb/d             | – 0.088             |                |             |
+---------------------------------------------------+-------------+--------------------------+---------------------+----------------+-------------+

Table: Throughput and read lengths of different sequencing technologies. \*: Normalized throughput is scaled to a one-hour period and rounded. \*\*: The throughput scale is compared to  Life Technologies 3730 Sanger chemistry based sequencer and shows the ratio of throughput values in terms of order of magnitude. \*\*\*: Numbers are based on vendor information: Illumina Inc. (www.illumina.com), Life Technologies (www.lifetechnologies.com), Roche/454 (www.454.com). Due to lack of information on sequencing statistics or commercial availability, Pacific Biosciences (www.pacificbiosciences.com), Oxford Nanopore Technologies (www.nanoporetech.com) and Helicos Biosciences (www.helicosbio.com) are excluded. {#tbl:bib_sequencing_technologies}

## Bioinformatic analysis of metagenome samples

NGS produces large volumes of sequence data ([@tbl:bib_sequencing_technologies]). Currently, a single run of an Illumina HiSeq machine generates up to 600 Gb per run (www.illumina.com), which is of the order of 104 times the amount of data produced in a similar timeframe by a Sanger sequencing chemistry based sequencer ([@tbl:bib_sequencing_technologies]). This, in turn, results in drastically increased runtimes for all the bioinformatics procedures applied in metagenomics [@WilkeningUsing2009], such as assembly of sequence fragments, taxonomic binning, prediction of protein encoding genes, as well as functional and process-level gene annotation. Together, taxonomic binning and assembly allow draft genome reconstructions for community members for which sequencing has recovered substantial amounts of sequence. Assembly corresponds to the computational process of placing individual reads into longer pieces of contiguous sequences, known as contigs, based on sequence overlaps and paired read information. Taxonomic binning sorts the contigs of a metagenome sample into ‘bins’ that represent the populations or higher-level clades of community members. Though both tasks are performed independently and evaluate different types of information, the problem of metagenome sequence assembly is closely related to taxonomic binning, as both allow the reconstruction of draft genome sequences. The terms “taxonomic” and “phylogenetic” binning are both used in the literature, as modern taxonomies such as the NCBI taxonomy [@SayersDatabase2009] or the ribosomal gene based RDP-II [@ColeRibosomal2009], GreenGenes [@DesantisGreengenes2006] and ARB-SILVA [@PruesseSilva2007] taxonomies are built upon phylogenetic principles. Even though it is less consistent, taxonomic binning software for shotgun metagenomics most frequently relies on the NCBI taxonomy, probably due to its widespread use in annotation of public sequence data.

Similar to the assembly of individual isolated genomes [@MillerAssembly2010], assembly in metagenomics aims to recover long contiguous pieces of sequence from the sequence collection of reads that represent parts of the genomes of individual community members. Massively increased amounts of data, varying organism abundances within a sampled community, differing complexities in terms of the overall number of organisms contained and the presence of multiple closely related organisms all challenge the sequence assemblers that were originally designed for isolated genomes. To address these challenges, methods designed for assembly of microbial community NGS data [@PellScaling2012a; @KorenBambus2011; @PengMetaidba2011; @LasersonGenovo2011] are being developed. Paired-end or mate-pair protocols, which add distance information between two individual reads, can greatly aid in the assembly process. Assembly information such as the ordering of contigs within a scaffold can also be used to check binning quality, and binning has been used to refine assembly in a feedback process. In recent studies, the joint analysis of assembly information and sequence composition allowed the reconstruction of several partial genomes by taxonomic binning [@HessMetagenomic2011; @IversonUntangling2012]. Thus, a closer integration of the two approaches appears promising for draft genome reconstruction from NGS metagenome data.

Following assembly and binning, further bioinformatic analyses include the prediction of genes, as well as functional annotation and reconstruction of potential pathways. For these steps, dedicated web servers exist, such as MG-RAST [@MeyerMetagenomics2008], IMG/M [@MarkowitzImg2012] and CAMERA [@SunCommunity2011]. Analysis of the gene content of individual bins allows inference of the functional and metabolic capabilities of individual community members, and allows a metagenome sample to be studied in its entirety. If read lengths or sequencing depth are insufficient for assembly, the functional analysis of a metagenome sample is restricted to what can be inferred without partial genome reconstructions for individual community members.

## Binning strategies

The term binning was originally coined for the problem of separating the sequence fragments of a metagenome according to the microbial populations they originate from [@WoykeSymbiosis2006; @TysonCommunity2004]. The definition has been extended to include bins that represent all fragments that originate from a common higher-level clade, in cases where resolution down to individual populations is not possible. For placement of sequence fragments into taxonomic bins, attributes which are indicative of the taxonomic origin of a fragment are evaluated. Different types of information can be used for this purpose: (a) local sequence similarity to sequences of known taxa (used in similarity-based taxonomic assignment), (b) similarity in sequence composition to sequences of a given taxon (used in composition-based taxonomic assignment) or to other sequences in the sample (used in composition-based clustering), or (c) similarity in read coverage and linkage information from assembly for contigs within a metagenome sample. The underlying rationale of using read coverage is that similar coverage of two contigs in the sample indicates similar abundance and therefore potentially the same underlying source population in the community.

How accurately fragments can be assigned to taxonomic bins depends on several factors. The first is fragment length. Shorter, noisier fragments cannot be assigned as accurately as longer fragments of 2 kb or more [@PatilTaxonomic2011].  In particular, assignment of individual reads or of fragments below 1 kb in length poses significant challenges. Reported assignment accuracies for 100 bp fragments to a clade at the genus level are 60% under somewhat idealized conditions, with only reference data from the same species being removed. This, however, means that 40% of fragments are misassigned [@BradyPhymm2009]. Furthermore, accuracy drops to less than 30% if the reference data is depleted of sequences from the same genus, meaning 70% of 100 bp fragments are misassigned at the family level.

Another influential factor for binning accuracy is the community’s complexity in terms of the number of distinct phylotypes it comprises. Metagenome sequencing of complex communities, such as those found in soil [@MackelprangMetagenomic2011], results in lower sequencing coverage of most populations and therefore shorter contigs in assembly. This amounts to many short fragments, or even predominantly unassembled samples, which have to be separated into a multitude of taxonomic bins. The larger the number of bins, the harder the problem becomes, as the chances of randomly assigning a fragment correctly decrease with increasing numbers of bins. Finally, for taxonomic assignment, the availability of reference data from taxa that are closely related to the microbes of the sequenced community is important for accurate assignment. Similarity-based assignment of metagenome shotgun sequence data requires homologous reference sequences from related taxa to be available for a fragment to be assigned; ideally, entire sequenced genomes should be available. The sequencing of many isolate genomes of the human microbiome in the Human Microbiome Project has immensely helped similarity-based taxonomic assignment of human gut metagenome samples [@QinHuman2010; @NelsonCatalog2010]. A ‘shallow’ (i.e. to high-ranking clades only) taxonomic assignment of a sample based on sequence similarities indicates the presence of many taxa that are only distantly related to isolated sequenced genomes. If no sequenced genomes from related taxa are available, composition-based assignment can be used for higher resolution taxonomic binning. Clustering of metagenome fragments based on sequence composition does not require reference sequences and comparably small amounts of non-homologous reference sequences are required for composition-based taxonomic classification.

+---------------------------+----------+----------+----------+------------------------------------------+
| Name                      | Phylo\-  | Tax.     | Funct.   | Techniques &                             |
|                           | typing   | assign\- | anno\-   | web link                                 |
|                           |          | ment     | tation   |                                          |
+:==========================+:========:+:========:+:========:+:=========================================+
| CAMERA v.2                | X        | —        | X        | Reverse Psi-BLAST                        |
| [-@SunCommunity2011]      |          |          |          | http://camera.calit2.net                 |
+---------------------------+----------+----------+----------+------------------------------------------+
| MetaABC                   | —        | X        | —        | BLAST, PhymmBL, MEGAN, Sort-ITEMS        |
| [-@SuMetaabc2011]         |          |          |          | http://bits2.iis.sinica.edu.tw/ MetaABC/ |
+---------------------------+----------+----------+----------+------------------------------------------+
| MG-RAST v.3.1.2           | X        | —        | X        | BLAST/BLAT                               |
| [-@MeyerMetagenomics2008] |          |          |          | http://metagenomics.anl.gov              |
+---------------------------+----------+----------+----------+------------------------------------------+
| MLTreeMap v.2.06.1        | X        | —        | X        | BLAST, HMMER, RaxML                      |
| [-@StarkMltreemap2010]    |          |          |          | http://mltreemap.org                     |
+---------------------------+----------+----------+----------+------------------------------------------+
| NBC v.1.1 CLI             | —        | X        | —        | Naïve Bayesian Classifier                |
| [-@RosenNbc2011]          |          |          |          | http://nbc.ece.drexel.edu                |
+---------------------------+----------+----------+----------+------------------------------------------+
| PhyloPythia               | —        | X        | —        | (Structured) Support Vector Machine      |
| [-@MchardyAccurate2007],  |          |          |          | http://binning.bioinf.mpi-inf.mpg.de     |
| PhyloPythiaS              |          |          |          |                                          |
| [-@PatilTaxonomic2011]    |          |          |          |                                          |
+---------------------------+----------+----------+----------+------------------------------------------+
| TaxSOM                    | —        | X        | —        | Self-Organizing Maps                     |
| [-@WeberPractical2011]    |          |          |          | http://soma.arb-silva.de                 |
+---------------------------+----------+----------+----------+------------------------------------------+
| WebCARMA v.3.0            | X        | X        | X        | BLAST, HMM search versus Pfam            |
| [-@GerlachTaxonomic2011]  |          |          |          | http://webcarma.cebitec.uni-bielefeld.de |
+---------------------------+----------+----------+----------+------------------------------------------+

Table: Overview of existing web applications for taxonomic assignment and phylotyping of metagenome sequence samples. Phylotyping methods assign only a subset of contigs based on taxonomic marker genes. {#tbl:bib_binning_web}

## Taxonomic binning based on sequence similarities

Similarity-based taxonomic assignment utilizes the local similarity of a query sequence to sequences of known taxonomic origin. Taxonomic identifiers are commonly assigned either by identifying the lowest common ancestor (LCA) from the taxonomy for the taxa of the most similar sequences found [@PatilTaxonomic2011] or by using phylogenetic placement methods. Phylogenetic placement methods, such as pplacer [@MatsenPplacer2010], EPA/RaxML [@BergerPerformance2011] and SEPP [@MirarabSepp2012] place the query sequence within a fixed reference tree. The taxonomic label assigned then corresponds to the LCA of the taxa associated with the first ancestral node's children. Both methods are related to ‘nearest neighbor’ classification. In both cases, there has to be a search phase in which such similarities are identified. Typically, local similarities to sequence database entries are searched for with alignment programs such as BLAST [@CamachoBlast2009]. Searches for gene family or protein domain motifs in the query sequence can be performed with a reference collection of profile Hidden Markov Models (HMMs). HMMER 3.0, released in 2010, has a 100-fold increase in speed compared to prior versions, with runtimes being competitive to blastp [@FinnHmmer2011]. Screening a large metagenome sample with a collection of profile HMMs for marker genes is computationally much less demanding than a full search for similar regions in large sequence collections [@FinnHmmer2011]. This is because the number of entries to be searched against is typically several orders of magnitude lower. HMMs are popular in combination with phylogenetic placement approaches, as the required multiple alignment of a query sequence to the homologs can be directly deduced from the state path of the sequence through the HMM and the multiple alignment used in its construction. However, known marker genes or protein families from reference collections such as PFAM only cover a small part of the genes found across diverse environments. Therefore, most HMM-based approaches [@StarkMltreemap2010; @GerlachTaxonomic2011; @WuPhylogenomic2012] may be seen as phylotypers of metagenome samples, rather than binning methods, as they indicate the taxonomic composition of the sample based on placement of a fraction of the fragments, rather than assigning the entire sample.
Searching for similar sequences in large sequence collections results in a higher fragment coverage with hits than when profile HMMs are used. Analysis of a metagenome sequence sample therefore comes with high computational costs, beyond what a typical desktop computer is capable of. When using a similarity search, one is therefore confronted with the question of which reference sequences to compare to. The choice depends on the available time and computational resources. Databases that are often searched are NCBI RefSeq, a non-redundant nucleotide and protein collection for medical, functional and diversity studies; NCBI whole genomes; NCBI nt, a large nucleotide collection; and NCBI nr, a large non-redundant protein collection [@SayersDatabase2009]. Software such as MEGAN [@HusonIntegrative2011] allows the output of BLAST to be interpreted for the taxonomic and functional characterization of metagenome samples based on sequence similarity. If sequenced genomes of related species to the sampled taxa exist, recruitment analysis has been used [@QinHuman2010]. Here, each read is compared to a set of genome sequences and 'recruited' to the most similar genome, allowing the identification of reads of the prevalent species that are closely related to a sequenced reference collection, if performed with stringent alignment cut-offs [@XieCommunity2010].

### Case study 1: Recruitment analysis

*In [@XieCommunity2010], Illumina and Roche/454 sequencing were jointly used to generate 860 Mb of non-human sequence data from a microbial community of human dental plaque. All obtained reads were aligned against 50 available reference genomes for human oral microbes from the Human Microbiome Project using Mummer, resulting in recruitment of 4% of all reads with more than 97% sequence identity to one of the reference genomes. This indicates that most of the sampled microbes originate from species that are too distantly related to the sequenced reference collection for similarity-based recruitment.*

## Taxonomic binning based on sequence composition

The composition-based approach to taxonomic binning is to utilize the taxonomic signal contained in fragment-wide GC content, codon usage or the use of short oligomers (kmers), typically 4–6 bp long.  The observation that such properties tend to vary more across the genomes of different species than within a given one gave rise to the term genome signatures [@KarlinDinucleotide1995; @DeschavanneGenomic1999]. Such signatures can also be inferred for higher-level clades, allowing their use for taxonomic fragment assignment across various ranks [@MchardyAccurate2007].

Taxonomic binning based on sequence composition can be performed with supervised or unsupervised methods. The choice of which to use depends on the availability of suitable reference data. Unsupervised methods group fragments with similar composition profiles into clusters, corresponding to individual taxonomic bins. Inference of the taxonomic label for a bin can be performed based on taxonomic assignment of marker genes found in the fragments of a bin. To infer the clustering of fragments, existing methods use, for example, a graph-cut algorithm or variations of a self-organizing map algorithm [@ChatterjiCompostbin2008; @WeberPractical2011]. A sample can also be binned with supervised methods, which assign fragments to clades using a model trained with available reference sequences. Supervised methods tend to have higher accuracy than unsupervised methods for taxonomic assignment and are more easily applied to complex microbial mixtures with skewed organism abundances. However, they require sufficient amounts of reference sequences to be identified for the sample populations or higher-level clades which are to be included in the model. In practice, therefore, each approach has its own appeal and both are being applied. Methods used for supervised classification are, for example, (structural) Support Vector Machines (SVMs) [@MchardyAccurate2007], the naïve Bayes classifier [@RosenNbc2011], a k-nearest neighbor classifier [@DiazTacoa2009] and Interpolated Markov Models [@BradyPhymm2009]. As composition-based signatures are a global attribute of sequences, no entire reference genomes are required, but only sufficient amounts of sequences for inference of a composition-based signature. For SVM-based classification, this has been found to be around 100 kb per clade [@PatilTaxonomic2011]. Reference sequences can be identified among publicly available genomes or by taxonomic assignment of conserved marker-genes of the sample contigs, which allows the respective contigs to be used as training material. If necessary, fosmids carrying marker genes can be sequenced to generate training material for interesting sample populations or higher-level clades [@WarneckeMetagenomic2007; @PopeAdaptation2010; @PopeIsolation2011].

### Case study 2: Taxonomic binning by composition-based taxonomic assignment

*In [@PopeAdaptation2010], a microbial gut community from the Australian Tammar wallaby was studied by Sanger and 454 sequencing of metagenome plasmid and fosmid libraries. This microbial community is involved in the breakdown of plant biomass consumed by the host animal. Using 16S rRNA analysis, 236 distinct phylotypes were observed. Of the 16S rRNA sequences, 9% originated from a novel species, Wallaby group 1 (WG-1), in the family of Succinivibrionaceae. PhyloPythia, a composition-based taxonomic classifier, was used to train a model including the WG-1 and other relevant clades for species present in the community. Composition-based taxonomic assignment of the metagenome sample recovered a 2 Mb draft genome for WG-1. Metabolic reconstruction based on the draft genome allowed the cultivation requirements for WG-1 to be deduced, leading to isolation, characterization and a draft genome sequence for the previously unknown species. It also resulted in the finding that WG-1 contributes to the low-methane emission phenotype of plant biomass degradation in the Tammar wallaby.  The draft genome sequences from the isolate culture showed 98.9% sequence identity to the WG-1 metagenome bin, and 90% of shared reads and assemblies, indicating accurate reconstruction of the draft genome from the metagenome sample by composition-based taxonomic binning.*

## Hybrid methods

Several methods combine different types of information to improve predictive accuracy [@HessMetagenomic2011; @IversonUntangling2012; @BradyPhymm2009; @HusonIntegrative2011]. For instance, read coverage is combined with an analysis of kmer frequencies in clustering of fragments [@HessMetagenomic2011; @TysonCommunity2004]. Searches for similar sequences and analysis of linkage information from an assembly are also combined with composition-based taxonomic assignment, if the computational burden can be borne. This has particular advantages for short fragment analysis. Kmer signatures for fragments below 1 kb in length, particularly those of individual reads, are noisy, even more so than taxonomic conservation of sequence similarities [@PatilTaxonomic2011].

### Case study 3: Taxonomic binning based on clustering by sequence composition and read coverage

*In one of the most in-depth metagenome studies of a particular environment undertaken so far, 286 Gb of paired-end Illumina sequence reads were generated from a sample of the plant-fiber adherent microbiome from a cow rumen [@HessMetagenomic2011]. Rarefaction analysis of 16S rRNA indicated the presence of ~1000 distinct OTUs. Clustering of assembled contigs by agglomerative hierarchical clustering, based on tetramer frequencies and read coverage, resulted in the formation of 466 taxonomic bins. Fifteen of these were estimated to represent largely complete genomes (between 60% and 92%), based on their association with fully sequenced genomes from their respective clades. This estimate was based on the presence of a minimal set of core genes found in all sequenced genomes from the respective phylogenetic order.*

### Case study 4: Taxonomic binning based on assembly information and sequence composition [@IversonUntangling2012].

*SOLID sequencing of two marine samples generated 58.5 Gb of mate-paired reads of 50 bps in length. The number of phylotypes observed with16S rRNA analysis was not specified in detail; however, family-level taxonomic groups were observed with abundances of less than 10%. From the metagenome data, 300 Mb of contigs were assembled. Scaffolds – linked sets of contigs assumed to originate from one genome – were generated by splitting the assembly graph, which links contigs based on mate-pair information, according to mate-pair linkage scores, read coverage and tetranucleotide usage. Scaffold clustering by tetranucleotide usage generated 14 partial genome reconstructions from the two samples, for populations ranging in abundance from 4% to 10 % each in one of the samples. Reassembly of 11 mate-pair connected scaffolds that are binned together based on similar tetranucleotide statistics and manual gap closure allowed the recovery of a closed circular 2 Mb genome from an uncultured group, the marine group II Euryarchaeota.*

## Advantages and disadvantages of different binning approaches

Which binning methodology to use depends on multiple factors, such as the complexity of the analyzed microbial community, available reference sequences and computing ressources. For taxonomic assignment of arbitrary sequence fragments to a particular species based on sequence similarity, completely sequenced reference genomes of closely related taxa are ideally required, which are often not available. If no reference data exists for the species of the metagenome sample, homology-based taxonomic assignment to higher-level clades is more accurate than composition-based taxonomic assignment for short fragments of 1 kb or less [@PatilTaxonomic2011]. This length corresponds to individual reads with most sequencing technologies. The assignment of individual reads in general is, however, notably less accurate than assignment of longer fragments.

The runtime of sequence similarity searches increases proportional to the product of the metagenome sample size (number and length of contigs) and the size of the reference sequence collection. This makes it a computationally very demanding task for next-generation sequencing data sets. The required computing resources are not available in many experimental laboratories. If researchers are willing to submit their data to external facilities, data can be processed by web servers such as MG-RAST, IMG-M or CAMERA, which offer their computational resources to the community.

The choice of whether to cluster or classify based on sequence composition depends on availability of some reference data to train a composition-based classifier. Classification is likely to be more accurate than clustering in taxonomic assignment. However, if no reference data is avaible, clustering will allow resolution of taxonomic bins which otherweise would go undetected. If multiple types of information are included into the binning process, like it is done in hybrid approaches, this is likely to increase the overall amount and accuracy of assignments. Composition-based taxonomic assignment requires less reference sequences than homology-based assignment. This is because sequence composition is a globally conserved property, while sequence similarity depends on local sequence conservation between a query and target. Training times of a composition-based taxonomic classifier depend on the method used, but it requires typically considerably less time than searching a reference sequence collection. Once a composition-based model for taxonomic classification has been trained, execution times for classification again typically scale linearly with the metagenome sample size and are independent of a reference sequence collection. For composition-based clustering, no training phase is needed.  The runtime of clustering typically scales at least quadratically with the sample size, as it often involves pairwise comparisons.

## Future directions

The recent developments in sequencing technologies have considerably pushed the boundaries in terms of what can be learned from metagenome sequence samples. The high sequencing depth of microbial communities, in combination with the application of sophisticated algorithms, has allowed the retrieval of near-complete draft genomes from the metagenomes of many microbial communities, including highly complex ones, such as those found in soil [@MackelprangMetagenomic2011]. However, the size and heterogeneity of the different data types produced by the various novel techniques have created new challenges, which remain to be addressed. A prominent one is how to further reduce the computational requirements of searching for local similarities between giga- or even terabase-sized sequence samples and equivalently large reference sequence collections. Secondly, it remains to be explored how taxonomic assignment accuracy can be further improved for the vast majority of microbial community members that are only distantly related to sequenced isolate genomes. Due to the value of available sequences from related taxa for the taxonomic binning of a particular sample, efforts such as GEBA might help in this regard [@WuPhylogenydriven2009]. The GEBA project aims to construct a “Genomic Encyclopedia for Bacteria and Archaea" by strategic sequencing of microbial genomes from all major and minor taxonomic groups. As the cost of sequencing has decreased, partial genome reconstruction by single-cell genome sequencing is an attractive option for obtaining reference sequences for taxonomic binning and draft genome reconstruction [@WoykeOne2010] from metagenomes. Here, an individual cell from a microbial population within a community is isolated using techniques such as optical tweezers, fluorescence assisted cell sorting and others, and is then lysed and its genome sequence amplified with multiple displacement amplification prior to random shotgun sequencing.

Advances in single-molecule sequencing technologies now allow longer reads to be generated than what was possible using traditional Sanger sequencing. Even though this promises to resolve several issues associated with short read analysis, such as high error rates in binning, assembly and functional annotation, the larger sequencing error of some of these technologies, currently estimated to be around 15%, presents a different substantial hurdle. Therefore, assessing technology-specific errors and developing technology-specific denoising procedures, such as have been developed for 454 amplicon data [@QuinceAccurate2009], will be prerequisite to leveraging the value of these techniques for metagenome research.
An interesting research direction is to investigate whether composition-based binning is applicable for the analysis of samples with both microbial and viral content. Composition-based taxonomic binning has been successfully applied for the analysis of viral metagenome samples, however, bacteriophage codon usage to some extent reflects properties of the host [@PrideGenome2008; @LucksGenome2008]. Therefore, classification accuracy and level of taxonomic resolution attainable for viral taxa will have to be investigated in more detail.

## Summary of key points

NGS technologies generate massive amounts of sequencing data allowing the in-depth analysis of microbial communities. Taxonomic binning has allowed draft genomes of microbial species from many environments to be reconstructed, and the cultivation requirements of a novel uncultured species to be deduced. To further advance draft genome reconstruction from metagenome samples, the existing techniques could be further refined by integrating multiple sources of information and by appropriately denoising the data under consideration to remove technology-specific sequencing errors.

## Acknowledgements

We thank Chris Quince and Alex Scyrba for providing comments.

## Funding

This work was supported by the German Max Planck society and Heinrich-Heine University Düsseldorf.

## References
