### Taxonomic Binning of Metagenome Samples (Review) {#sec:publication_bib}

|  |  |
| :---------- | ------------------------------------------------------------ |
| **Title** |  Taxonomic binning of metagenome samples generated by next-generation sequencing technologies |
| **Journal** |  Briefings in Bioinformatics |
| **Published** | 31 July 2012 |
| **Authors** | Johannes Dröge, Alice C. McHardy |
| **DOI** | [10.1093/bib/bbs031](https://doi.org/10.1093/bib/bbs031) |

![BiB Review Article Preview](figure/publication_bib_preview.png){#fig:publication_bib_preview}

#### Abstract

Metagenome research uses random shotgun sequencing of microbial community DNA to study the genetic sequences of its members without cultivation. This development has been strongly supported by improvements in sequencing technologies, which have rendered sequencing cheaper than before. As a consequence, downstream computational analysis of metagenome sequence samples is now faced with large amounts of complex data. One of the essential steps in metagenome analysis is reconstruction of draft genomes for populations of a community or of draft ‘pan-genomes’ for higher level clades. ‘Taxonomic binning’ corresponds to the process of assigning a taxonomic identifier to sequence fragments, based on information such as sequence similarity, sequence composition or read coverage. This is used for draft genome reconstruction, if sequencing coverage is insufficient for reconstruction based on assembly information alone. Subsequent functional and metabolic annotation of draft genomes allows a genome-level analysis of novel uncultured microbial species and even inference of their cultivation requirements.

#### Introduction

#### Next-generation Sequencing Technologies

#### Bioinformatic Analysis of Metagenome Samples

#### Binning Strategies

#### Advantages and Disadvantages of Different Binning Approaches

#### Future Directions
