## Metagenome binning

Biotechnological and medical applications are the major drivers to search for new enzymes [@missing_citation]. The detailed study of protein-coding genes and their regulation requires more than a nucleotide gene sequences or a simple list of species. The corresponding genomes carrying the target genes need to be isolated. The sorting of physical cells is difficult as there may be hundreds or thousands of OTUs in a metagenomic sample [@HessMetagenomic2011; @WoykeOne2010; @WoykeAssembling2009]. Even then, the specifically required cultivation conditions to produce clone libraries are usually unknown and direct sequencing may not be possible due to the limited amount of input DNA. For these reasons, in-silico methods provide a good alternative to physical separation. Metagenome sequence binning is the algorithmic approach for (partially) reconstructing individual genomes from shotgun metagenome sequence data. A genome bin is simply a set of sequences, usually contigs, which all-together are believed to present the sequenced part of a community genome.

### Supervised vs. unsupervised binning methods

There are many different algorithmic approaches to infer genome bins and most of them are either a form of data clustering or classification (though hybrid methods exists). The major difference between these two approaches is that clustering is a so-called unsupervised method which does not directly take into account external information, for instance available genome sequences. The strength of clustering is that it can group any data to explore their intrinsic structure, for instance to group contigs of genomes which have never been seen before. In contrast, classification algorithms utilize categorized data, for instance large genome sequence collections, to assign sequences to genome bins. They are said to operate in a supervised manner. Due to the use of additional information, they can be very efficient but a major drawback is the difficulty to handle novel genomes that have not been observed before. Since clustering and classification complement each other, it is common to combine them in one or another way, for instance classifying genome bins after clustering or initializing clusters using classification labels [@ImelfortGroopm2014].

### Sequence features

Both clustering or classification methods for binning operate on sequence (contig) features. These are properties which inform about genome membership and discriminate contigs of different genomes. Genomes expose a characteristic frequency distribution of short nucleotide motifs [@missing_citation:genome_signature_see_revision] which is generally called the nucleotide composition and often used in binning. The simplest form to measure nucleotide composition is the combined relative frequency of guanine and cytosine nucleotides (GC-content), an evolutionary trait of genomes that has long been used to characterize different species. For instance, most Actinobacteria expose a high GC-content [@missing_citation:actinobacteria_high_gc] which has been associated with robustness to high temperatures [@missing_citation]. Most methods, however, use short nucleotide motifs consisting of 4 to 7 bases called $k$-mers ($k$ stands for the number of bases). Alternative formulations may use Hidden Markow Models (HMMs) to describe nucleotide composition [@BradyPhymm2009]. A second major information type for binning is the amount of sequencing reads covering each assembled contig. Read assembly works by aligning reads which overlap with others so that each nucleotide position of a resulting contig must be covered by at least a single read, but typically many more. Following random shotgun sequencing with universal primers, the expected number of reads covering a single position is approximately proportional to the genome copy number in the sequenced sample, with a constant factor which depends on the total sequencing effort. Contig coverage information thus helps to discriminate genomes with differing sample abundance but cannot differentiate between equally abundant genomes. Hence it is desirable to generate multiple samples for which the genome copy numbers vary differently so that each genome has unique genome abundances for binning. This requires a corresponding experimental setup before the metagenome of the microbial community is sequenced. Recent studies have shown that genome abundances represent a very informative feature type to obtain genome bins for complex metagenomes, if many varying samples are available [@AlnebergBinning2014]. Rarely, binning programs may also employ assembly information such as associated contigs which are linked by paired reads or scaffolding (a process using the same input as assembly) [@LuCocacola2016]. Such information, if available, is more often used to link, quality-check or refine genome bins after binning [@PatilTaxonomic2011].

There is a specific class of homology-based classifiers, and an example of such a method is described in [@sec:publication_taxator-tk]. These methods employ a two-step procedure, first identifying potential homologs for a contig, for instance by alignment to reference sequences, and second determining a corresponding evolutionary neighborhood. This neighborhood is usually reported by taxon so that each contig is annotated with a taxonomic path. When contigs are grouped by taxa and those paths lead down to the species or strain level, then these groups can be viewed as incomplete genome bins. However, higher-level taxon bins mix contigs from several genomes, if the sample contains more than one member of this clade. Hence, taxonomic classification using sequence similarity only represents a partial solution to the binning problem. However, taxonomic annotation also informs about the phylogenetic sample composition and diversity, similar to a 16S analysis, and may be used as secondary features for unsupervised methods, for instance to initialize genome clusters [@ImelfortGroopm2014] or to train a classification model with sample data [@GregorPhylopythias2016]. The binning framework presented in [@sec:publication_mglex] utilizes taxon annotation as features for binning, similar to nucleotide composition and contig coverage information.

### Binning programs

As binning programs emerged and evolved together with metagenome sequencing and assembly, their focus changed according to current protocols. Recent programs target long (1 kb or more) assembled contigs but earlier programs were also designed to bin raw sequencing reads [@missing_citation:read_binner], for instance by comparison to genome sequence collections or nucleotide composition. Since the latter proved unstable for sequences of only a few hundreds of bases [@MchardyAccurate2007], these programs were inherently limited to simple communities and community members which had related genome sequences available for comparison. Most newer binning programs with applications to real metagenomes in [@tbl:binning_programs] focus on co-assembled contigs which were constructed using multiple sequenced samples.

| Program | Technique | Sequence information | Published/ update | License |
| :--------------- | :-------------------- | :------------------------- | :--------- | :---------- |
| PhyloPythiaS | Structured Support Vector Machine (SVM) | 5-mers | 2011/2012 | proprietary |
| MetaWatt 3.x | Heuristic thresholds | 4-mers, differential coverage | 2012/2015 | AFL |
| CONCOCT | Gaussian mixture clustering | 4-mers, differential coverage | 2014/2015 | BSD |
| GroopM | Biclustering | 4-mers, differential coverage | 2014/2016 | GPL |
| MaxBin 2.0 | Expectation-Maximization (EM) clustering | 4-mers, differential coverage | 2014/2016 | BSD |
| MetaBAT | Distance-based clustering | 4-mers, differential coverage | 2015/2016 | proprietary |
| PhyloPythiaS+ | Structured Support Vector Machine (SVM) | 5-mers | 2016/2014 | proprietary |
| MyCC | Stochastic neighbor embedding | 4-mers, differential coverage | 2016/2015 | proprietary |
| COCACOLA | Gaussian mixture clustering | 4-mers, differential coverage, genome co-alignment, paired reads | 2016/2016 | GPL |

Table: Contig binning programs with methodology with release dates starting from the year 2011 up to the year 2016. This is a non-exhaustive list with rough methodology descriptions. Some programs employ additional sequence information in post-processing procedures which are omitted here. A good overview of binning methods is given in [@sec:publication_bib] and [@SedlarBioinformatics2017]. {#tbl:binning_programs}

### State of the art

Clearly, an optimal binning would mean to obtain a single bin for each genome in the community. Suboptimal solutions may contain either multiple smaller bins for a genome or bins with contigs that belong to different genomes. While the original objective is indisputable, it is impossible to obtain the optimal partition for real metagenome data if there is simply not enough information to discriminate the contigs, especially the shorter ones. There are many binning methods which all produce suboptimal bins and there is no clear agreement in the metagenomics community of how to rate and compare the bins obtained by different methods. Initiatives such as the [Critical Assessment of Metagenome Interpretation (CAMI)](http://cami-challenge.org/) work towards establishing a common understanding to judge metagenome binning. Different opinions exist, as the binning quality is often determined by its utility in downstream processing, which in turn depends on the specific research questions. For instance, the estimation of community structure might require an equally biased construction of adequately sized bins whereas someone interested in a few species genomes will probably tolerate a certain fraction of pollution in the corresponding bins and simply neglect the contigs contained in other bins. Therefore, binning methods are best judged in the context of their use cases.

Binning algorithms are also sensitive to the nature of the data so that a method which appears to work well for one ecosystem might largely fail for another one because it is confronted with a completely different metagenome with regard to the number and abundances of genomes and their sequence content. This problem is most pronounced for methods which represent instances of classification algorithms, for which external genome sequence data is used to group the sample contigs. Metagenomes with medical applications, for example human gut microbiota, are usually well suited to the classification approach because there are abundant genome data for microbes related to human health. On the contrary, the analysis of a deep sea vent community is likely too exotic for plain classification.

The broad range of applications involving many different microbial habitats, custom experimental techniques and heterogeneous sequencing platforms makes it difficult to define a state of the art for binning. Nonetheless, general trends can be observed. Recent works which present genome bins derived from complex metagenomes often applied clustering in concatenated and transformed feature spaces [@ImelfortGroopm2014; @AlnebergBinning2014; @KangMetabat2015; @LinAccurate2016]. These multi-dimensional spaces integrate several types of features including nucleotide composition and contig coverage for multiple samples. However, the derivation of high-quality draft genomes still relies on custom analysis and processing of genome bins [@AlbertsenGenome2013; @ErenAnvi2015].
