\newpage

## Conclusions and outlook

Metagenomics as a discipline has matured in the course of this thesis, with regard to nucleotide sequencing, metagenome assembly and computational analysis. For instance, paired read insert libraries and long-read technologies allow assembling larger fractions of metagenomes. The subsequent assembly of metagenomes is considered an important task, which is different from isolate genome assembly leading to the development of dedicated algorithms. The interest in medical applications has been continuously increasing in metagenomics so that the analysis of microbial communities, such as the human gut microbiome, and its impact on human health is on the way to become a standard procedure. Several large-scale projects reflect this increased interest, for instance the [Human Microbiome Project (http://hmpdacc.org)](http://hmpdacc.org/), [MetaHIT (http://www.metahit.eu)](http://www.metahit.eu/) or the [Earth Microbiome Project (http://www.earthmicrobiome.org)](http://www.earthmicrobiome.org/), which all seek to collect data using standardized protocols and analysis methods.

For the analysis of metagenomics data, the impact of algorithms on overall study conclusions may not be underestimated, as most of the data is directly or indirectly produced by computer programs. Each specific analysis procedure is sensitive to the applied software pipeline and the results may, for example, differ in the number and abundances of OTUs, the quality of assembled genome sequences and the robustness to particular experimental details such as sequencing errors. For the multitude of methods which have been developed over the past years, including the methods presented in this thesis, it is still to determine under which conditions they should, or should not, be applied and how they compare to other methods which claim to solve similar problems. Therefore, in addition to developing new methods, rigorous testing is required to provide a more complete picture of the metagenomic software landscape for the scientific community. An example of an effort to improve on metagenomic method comparison is the CAMI challenge. During our work on software evaluations for the CAMI initiative, we noted that software accessibility represents an important factor, among others like code quality and program usability, in order to enable systematic testing and reproducibility of results.

This thesis and the methods presented here contribute to the field of metagenomics by providing some generic base-level metagenome analysis tools. They implement new theoretical approaches and are accessible for evaluation and application as open-source. Both *taxator-tk* and *MGLEX* are also suitable to assess the quality of metagenome assemblies and binning from various environments. The methods contribute on the path towards fully automatic recovery of high-quality genomes. This target may be reached in the near future, not only by algorithmic improvements in metagenomics but also in combination with new experimental techniques and progress in sequencing technology. On one side, single-molecule sequencing will eliminate problems in metagenome processing, which are associated with the short read length. On the other side, single cell sequencing [@BremgesMecors2016; @MendeImproved2016] allows assembling genomes from very limited numbers of microbial cells that need to be isolated but not grown in medium [@LaskenRecent2014; @GawadSinglecell2016].
