---
abstractenglish: |
 | Metagenomics is dedicated to the analysis of microbial communities. These communities may consist of thousands upon thousands of microorganisms, including bacteria, archaea, algae and fungi, which coexist within their habitats but which cannot simply be cultivated and studied due to their complex mutual dependencies and specific environmental requirements. Microbial communities can be found in almost every place, from biogas reactors over deep sea vents, the surface of plant leaves and roots, to the human body, which hosts a plethora of foreign cells in its digestion system. These ecological niches are colonized by species, whose genes guarantee their survival, for instance by making nutrients accessible, by neutralizing toxic compounds or by allowing symbiosis with other organisms. Through the use of nucleotide sequencing technologies, this genetic diversity can be utilized, for instance in the form of new pharmaceuticals or as enzymes in biotechnology. Apart from its considerable economic potential, metagenomic approaches lead to a fundamentally improved understanding of the microbial processes on earth.
 | With current technology, it is not directly possible to sequence contiguous genomes from microbial communities. Instead, short sequences, called reads, are produced, which need to be assembled into genes and longer genome sequences using computer programs. Depending on the size and complexity of the metagenome, this task can be very difficult. This thesis describes two methods for assigning metagenomic sequences to taxonomic groups or genomes in order to analyze the genes, and the corresponding proteins and functions, within their phylogenetic and genetic context to gain better insight into the functioning of individual organisms and the entire microbial community.
 | The first presented program, *taxator-tk*, solves the problem of taxonomic classification of nucleotide sequences from metagenomes. Common challenges are on the one hand the precise prediction of corresponding taxa and on the other hand the application to datasets, which are constantly growing due to the rapid progress in DNA sequencing. Annotation methods such as *taxator-tk*, which require similarity to known genomes, spend a considerable part of their runtime for sequence comparison. Thus, the algorithm in our method exploits the underlying phylogenetic structure for similar gene sequences to efficiently calculate the taxonomic assignment. The same phylogenetic principles are used to achieve a high assignment precision.
 | The second method, which is presented in my thesis, helps researchers to reconstruct individual genomes. It is a statistical classification model for metagenome data, for which we outline several direct and follow-up applications, including classification of nucleotide sequences to individual genomes, *de-novo* calculation of genome clusters in metagenomes, *in-silico* sample enrichment for genomes and quality checking of reconstructed genomes. Since such models can form the basis of higher-level procedures, it was published as a software library named *MGLEX*, which can be embedded into specific programs to enable the efficient use of the data for reconstructing genomes in different scenarios.
 | Presumably, metagenomics will continue to play an important role in microbial research, and may partially obviate the genome sequencing of cloned strains. This view is supported by the rapid development of DNA sequencing technologies, which is progressing towards faster sequencing and longer reads. The presented methods supplement the existing set of bioinformatics tools for acquiring knowledge from metagenomes, although they may come in useful in other areas. By reducing metagenomes to individual genomes, one can apply traditional algorithms from genomics, such as the reconstruction of metabolic pathways, and link data from established experimental techniques, for instance transcriptome sequencing and proteome analysis. Therefore, there is much interest in genome reconstruction methods, like the ones presented in this thesis.
...
