# Abstract

## Summary

Recovering genomes from metagenomic shotgun data is challenging: First, metagenomes are  heterogeneous like the underlying communities, containing a few to hundreds of species and strains. Second, shotgun data are diverse because they are produced by various experimental designs, sequencing technologies and software post-processing. We present a robust and modular probabilistic model with many features for genome recovery which can be embedded into programs and custom workflows. We validated the model with simulated short contigs of a complex community and found that it was consistent, assigning likelihood scores with p-values to each contig. We showcase soft genome classification, genome-targeted data enrichment and post-binning analysis using our software package MGLEX to demonstrate its potential and versatility for genome recovery.


## Zusammenfassung

Recovering genomes from metagenomic shotgun data is challenging: First, metagenomes are  heterogeneous like the underlying communities, containing a few to hundreds of species and strains. Second, shotgun data are diverse because they are produced by various experimental designs, sequencing technologies and software post-processing. We present a robust and modular probabilistic model with many features for genome recovery which can be embedded into programs and custom workflows. We validated the model with simulated short contigs of a complex community and found that it was consistent, assigning likelihood scores with p-values to each contig. We showcase soft genome classification, genome-targeted data enrichment and post-binning analysis using our software package MGLEX to demonstrate its potential and versatility for genome recovery.

