## Discussion

Metagenomics as a discipline has matured during the course of this thesis, with regard to nucleotide sequencing, metagenome assembly and computational analysis. For instance, paired read insert libraries and long-read technologies allow to assemble larger fractions of metagenomes. The assembly of metagenomes is considered an important task which is different from isolate genome assembly and for which dedicated algorithms were developed. The interest in medical applications has been continuously increasing so that the analysis of microbial communities, such as the human gut microbiome, and its impact on human health is becoming a standard procedure. The increased interest is reflected by several large-scale projects, for instance the Human Microbiome Project [website], MetaHIT project [website] or the Earth Microbiome Project [website], which seek to collect data using standardized protcols and analysis methods.

The impact of algorithms on the overall conclusions may not be underestimated, as most of the data is directly or indirectly produced by computational processing. Each metagenome analysis is sensible to the software pipeline used and the results may, for example, differ in the number and abundances of OTUs, the quality of assembled genome sequences and the robustness to different sequencing errors. For the multitude of methods which have been developed over the past years, including the methods presented in this thesis, it is still to determine under which conditions they should, or should not, be applied and how they compare to other methods which claim to solve similar problems. Therefore, in addition to developing new methods, rigorous testing is required to provide a more complete picture of the metagenomic software landscape for the scientific community. An example of an effort to improve on metagenomic method comparison is the CAMI challenge (Critical Assessment of Metagenomic Interpretation) [website/paper]. During our work on software evaluations for the the CAMI initiative, we noted that software accessability represents an important factor among others like code quality and program usability, in order to enable systematic testing and reproducability of results.

This thesis, and the methods presented here, contribute to the field of metagenomics by providing some generic base-level tools for metagenome analysis. These implement new theoretical approaches and are accessible for testing and application as open-source. Both *taxator-tk* and *MGLEX* are also suitable to assess the quality of metagenome assemblies and binning from various environments. The methods may come in handy on the path towards fully automatic recovery of high-quality genomes. This target may soon be reached, not only by algorithmic improvements in metagenomics but also by new experimental techniques, for instance single cell seqencing, which allows to assemble genomes from very limited numbers of microbial cells that need to be isolated but not grown in medium [cite].
