## Metagenome Binning

Biotechnological and medical applications are major drivers for the quest for new enzymes []. However, a detailed study of gene functions requires more than a simple list of species or gene sequences. The corresponding genomes carrying the targeted genes need to be isolated. Experimental cell extraction is difficult as there may be hundreds or thousands of OTUs in the sample. Even then, the specific cultivation conditions are usually not known and direct sequencing may not be possible with such a limited amount of input DNA. For these reasons, in-silico methods are mostly preferred. Metagenome sequence binning is the algorithmic approach to (partially) reconstruct individual genomes. A genome bin is simply a set of sequences, usually contigs, which all-together represent the sequenced part of a community genome.

There are many different algorithmic approaches to construct genome bins but most of them are either a form of data clustering or classification. The major difference here is that clustering is a so-called unsupervised method which does not directly take into account external information, for instance available genome sequences. The strength of these algorithms is that they can group any data to explore their intrinsic structure. In contrast, classification algorithms make use of categorized data, for instance large genome sequence collections, to assign sequences to genome bins. Due to the use of additional information, they can be very efficient but a major drawback is the difficulty to handle classes that have not been observed before, such as completely novel genomes. Since both approaches complement each other, it is common to combine them in one or another way, for instance to classify bins after clustering [] or to initialize clusters by classification results [].

Both clustering or classification solutions to binning operate on contig features. These are sequence properties which inform about genome membership or discrimination. Nucleotide composition is widely used for this purpose as genomes expose a characteristic frequency distribution of short nuclotide motifs [earliest genome signature paper]. This can be reduced to the GC content of a genome in the simplest case, a property that has long be used to characterize species genomes. However, most methods use sequences with 4 to 7 nucleotides called $k$-mers ($k$ stands for the number of nucleotides). Hidden Markow models (HMMs) can describe nucleotide composition [] in an alternative way. The second major feature type for binning is the amount of sequencing reads to cover each contig. Read assembly works by aligning reads which overlap with others so that each nucleotide position of a resulting contig must be covered by at least a single read (but typically many more). As an result of random shotgun sequencing with universal primers, the number of reads covering a single position is often proportional to the genome copy number in the sequenced sample. Contig coverage information thus helps to discriminate genomes with differing sample abundance but not those which are equally abundant. It is therefore desirable to generate samples for which genome copy numbers vary differently so that each genome has a unique vector of genome abundances for binning. Rarely, binning programs may employ further assembly information such as associated contigs which are linked by paired reads or scaffolding. However, such information, if available, can alway be used to quality-check or refine genome bins after binning [].

There is a special class of homology-based classifiers (and [publication 2: taxator-tk] describes an example of such a method). These methods employ a two-step procedure, first identifying potential homologs, for instance by alignment to reference sequences, and second determining a corresponding evolutionary neighborhood as output class. This neighborhood is usually reported by taxon so that each contig is annotated with a taxonomic path. When contigs are grouped by identical taxa and those paths lead down to the species or strain level, these groups can be viewed as (incomplete) regular genome bins but higher-level taxon bins can merge contigs from several genomes, if the sample contains more than one member of this clade. Hence, taxonomic classification using sequence similarity represents a partial solution to the binning problem. Furthermore, taxonomic annotation provides a good summary of the phylogenetic sample diversity and may be used as secondary features for further processing, for instance to initialize genome clusters [GroopM] or to train a classification model with sample data [PPS+].

(Methods paragraph)

| Program            | Technique                      | Type           | Sequence Features Used               | Year |
| :----------------- | :----------------------------- | :------------- | :----------------------------------- | ---: |
| TETRA              | Self-Organizing Map (SOM)      | unsupervised   | nucleotide composition (4-mers)      |    ? |
| PhyloPythia        | Support Vector Machine (SVM)   | supervised     | nucleotide composition (5-mers)      |    ? |

Table: Binning Programs {#tbl:binning_programs}
TODO: literature search

Since binning groups the available contigs by genome, the optimal solution is to obtain a single bin for each genome in the community. Suboptimal solutions may contain either multiple smaller bins for a genome or bins with contigs that belong to different genomes. While the original objective is indisputable, it is nearly impossible to obtain the optimum for real metagenome data because there is simply not enough discriminating information for the contigs, especially the shorter ones. There exist many binning methods which all produce suboptimal bins and there is no clear understanding in the metagenomics community of how to rate and compare the bins obtained by different methods [CAMI ref]. This seems natural as the binning quality for a community is often determined by its utility in downstream analysis which in turn depends on the research questions. For instance, the estimation of community structure might require an equally biased construction of adequately sized bins whereas someone interested in a few species will probably tolerate a certain fraction of extraneous genes in the corresponding bins and ignore the rest of the data. As a result, binning methods are best judged in the context of their use cases.

Binning algorithms are also sensitive to the nature of the data so that a method which appears good for some ecosystem might miserably fail for another one because it is confronted with a completely different metagenome with regard to the number and abundances of genomes and their sequence properties. This problem is most pronounced for methods which represent instances of classification algorithms for which external genome sequence data is used to group the sample contigs. Metagenomes with medical applications, for example the human gut microbiota, are often well suited to the classification approach because there is abundant genome data for microbes related to human health which can be used. On the contrary, the analysis of a deep sea vent community is likely too exotic to be suitable for simple classification.
