\newpage

### Further works {#sec:synopsis_furtherworks}

The published methods in this thesis were validated in a controlled manner, which usually involves data simulations and sampling from reference genome sequences. Eventually, when applied to a variety of real metagenomes, it must be shown that the demonstrated properties hold. The program *taxator-tk* was subsequently applied in two metagenome studies in completely different settings. In @BulgarelliStructure2015, taxonomic profiles were generated for metagenome contigs to study complex microbial communities associated with plant roots (rhizosphere). These profiles where shown to be consistent with community profiles based on independent 16S amplicon sequencing. Furthermore, *taxator-tk* was able to discover clades, for instance Archaea and Cyanobacteria, which the 16S primers seemed to miss in the amplification step. Such amplicon primer biases were also independently confirmed [@EloefadroshMetagenomics2016]. In a second study of a benzene-degrading enrichment community [@DongReconstructing2017}, *taxator-tk* was applied to derive bin-specific sequence data to train a full model for the composition-based classifier PhyloPythiaS [@PatilTaxonomic2011], so that the genomes of four species could be recovered, two of them with over 97% completeness. Thereby, we used the same logic for defining the model and for seeding the genome bins with training data as in the program PhyloPythiaS+ [@GregorPhylopythias2016], but we replaced the homology search based on marker genes with *taxator-tk*, which offered better coverage using more extensive reference data for this task. The completeness and potential contamination levels of the derived genomes were checked independently, based on single-copy marker genes. The near-complete genomes were subsequently used to study benzene degradation pathways by linkage to metabolomic experiments.

In the course of software evaluation and data processing, we learned that the current metagenome analysis toolbox features many programs which claim to solve similar problems but give different, sometimes misleading results. This is partly due to the fact that metagenomics is an interdisciplinary field with researchers from biotechnology, ecology and medicine, each with a different focus on ecosystems and data (see @fig:metagenomes_environments). As a result, metagenomics lacks a systematic and cross-discipline view on software for metagenome data processing and analysis. As part of the [Critical Assessment of Metagenomic Interpretation (CAMI) challenge (http://cami-challenge.org)](http://cami-challenge.org/), I participated establishing a common understanding by comparing computer programs for metagenome analysis, such as metagenome assembly, taxonomic profiling and genome binning [@SczyrbaCritical2017], both by taking part in the conception and implementation of the framework as well as by submitting a *taxator-tk* for comparison.
