## Metagenome Binning

Biotechnological and medical applications are major drivers for the quest for new enzymes []. However, a detailed study of gene functions requires more than a simple list of species or gene sequences. The corresponding genomes carrying the targeted genes need to be isolated. Experimental cell selection is difficult as there may be hundreds or thousands of OTUs in the sample. Even then, the specific cultivation conditions are usually not known and direct sequencing may not be possible with such a limited amount of input DNA. For these reasons, in-silico methods are mostly preferred. Metagenome sequence binning is the algorithmic approach to (partially) reconstruct individual genomes from shutgun metagenome sequences data. A genome bin is simply a set of sequences, usually contigs, which all-together represent the sequenced part of a community genome.

### Supervised vs. Unsupervised Methods

There are many different algorithmic approaches to infer genome bins and most of them are either a form of data clustering or classification (though hybrid methods exists). The major difference between these two approaches is that clustering is a so-called unsupervised method which does not directly take into account external information, for instance available genome sequences. The strength of clustering is that it can group any data to explore their intrinsic structure. In contrast, classification algorithms utilize categorized data, for instance large genome sequence collections, to assign sequences to genome bins. They are said to operate in supervised mode. Due to the use of additional information, they can be very efficient but a major drawback is the difficulty to handle novel genomes that have not been observed before. Since clustering and classification complement each other, it is common to combine them in one or another way, for instance classifying bins after clustering [taxator-tk, PhymmBL?] or initializing clusters using classifition labels [GroopM].

### Sequence Features

Both clustering or classification solutions to binning operate on sequence (contig) features. These are properties which inform about genome membership and discriminate different genomes. Nucleotide composition is widely used for this purpose as genomes expose a characteristic frequency distribution of short nuclotide motifs [earliest genome signature paper]. The simplest form to measure nucleotide composition is the GC content of a genome, a property that has long be used to characterize species. However, most methods use sequences with 4 to 7 nucleotides called $k$-mers ($k$ stands for the number of nucleotides). Alternatively, Hidden Markow Models (HMMs) can be used to describe nucleotide composition [Phymm]. Another major feature type for binning is the amount of sequencing reads covering each assembled contig. Read assembly works by aligning reads which overlap with others so that each nucleotide position of a resulting contig must be covered by at least a single read (but typically many more). As an result of random shotgun sequencing with universal primers, the expected number of reads covering a single position is proportional to the genome copy number in the sample, with a factor that depends on the total sequencing effort. Contig coverage information thus helps to discriminate genomes with differing sample abundance but cannot differentiate between equally abundant genomes. It is therefore desirable to generate multiple samples for which the genome copy numbers vary differently so that each genome has a unique vector of genome abundances for binning. This requires a corresponding experimental setup before sequencing the microbial community. Recent studies have shown that genome abundances represent the most informative feature type to obtain genome bins for complex metagenomes, if many varying samples are available [CONCOCT?]. Rarely, binning programs may also employ assembly information such as associated contigs which are linked by paired reads or scaffolding (a process using the same input as assembly). However, such information, if available, can also be used to quality-check or refine genome bins after binning [PPS].

There is a special class of homology-based classifiers (and [@sec:publication_taxator-tk] describes an example of such a method). These methods employ a two-step procedure, first identifying potential homologs for a contig, for instance by alignment to reference sequences, and second determining a corresponding evolutionary neighborhood as output label. This neighborhood is usually reported by taxon so that each contig is annotated with a taxonomic path. When contigs are grouped by identical taxa and those paths lead down to the species or strain level, groups can be viewed as (usually incomplete) regular genome bins. However, higher-level taxon bins can merge contigs from several genomes, if the sample contains more than one member of this clade. Hence, taxonomic classification using sequence similarity represents a partial solution to the binning problem. Furthermore, taxonomic annotation informs about the phylogenetic sample composition and diversity, and may be used as secondary features for further processing, for instance to initialize genome clusters [GroopM] or to train a classification model with sample data [PPS+].

### Binning Programs

(Methods paragraph)

| Program            | Technique                      | Type           | Sequence Features Used               | Year |
| :----------------- | :----------------------------- | :------------- | :----------------------------------- | ---: |
| TETRA              | Self-Organizing Map (SOM)      | unsupervised   | nucleotide composition (4-mers)      |    ? |
| PhyloPythia        | Support Vector Machine (SVM)   | supervised     | nucleotide composition (5-mers)      |    ? |

Table: Binning Programs {#tbl:binning_programs}

TODO: literature search

### State of the Art

Naturally, the optimal binning solution would mean to obtain a single bin for each genome in the community. Suboptimal solutions may contain either multiple smaller bins for a genome or bins with contigs that belong to different genomes. While the original objective is indisputable, it is nearly impossible to obtain the optimum for real metagenome data because there is simply not enough discriminating information for the contigs, especially the shorter ones. There exist many binning methods which all produce suboptimal bins and there is no clear understanding in the metagenomics community of how to rate and compare the bins obtained by different methods [CAMI ref]. This seems natural as the binning quality for a community is often determined by its utility in downstream analysis which in turn depends on the research questions. For instance, the estimation of community structure might require an equally biased construction of adequately sized bins whereas someone interested in a few species will probably tolerate a certain fraction of extraneous genes in the corresponding bins and ignore the rest of the data. As a result, binning methods are best judged in the context of their use cases.

Binning algorithms are also sensitive to the nature of the data so that a method which appears good for some ecosystem might largely fail for another one because it is confronted with a completely different metagenome with regard to the number and abundances of genomes and their sequence properties. This problem is most pronounced for methods which represent instances of classification algorithms for which external genome sequence data is used to group the sample contigs. Metagenomes with medical applications, for example the human gut microbiota, are usually better suited to the classification approach because there is abundant genome data for microbes related to human health. On the contrary, the analysis of a deep sea vent community is likely too exotic to be suitable for simple classification.

The broad range of applications involving many different microbial habitats, custom experimental techniques and hetergenous sequencing platforms makes it difficult to define a 'state of the art' for binning. Nonetheless, general trends can be observed. Recent works which present genome bins derived from complex metagenomes often applied clustering in concatenated and transformed feature spaces [MetaBat, CONCOCT, GroopM]. These multi-dimensional spaces integrate several types of features including nucleotide composition and contig coverage. Hence, metagenomic reads were assembled before binning because raw read binning seems impossible for complex communities. The binning success appears to depend strongly on the sampling and sequencing design as well.
